File: tool/table/client/weaponinfo.ts
/*
Author: 
Data: 
Desc: local data config
NOTE: Don't modify this file, it's build by xml-to-python!!!
*/

module config{

let weaponinfo_map = null;
export function weaponinfo_map_init(config_obj:Object):void{
	weaponinfo_map = config_obj["weaponinfo_map"];
}


export class Weaponinfo extends Object{
	private m_config:any;
	public static m_static_map:Object = new Object();
	constructor(key){
		super();
		this.m_config = weaponinfo_map[key];
		for(let i in this.m_config){
			this[i] = this.m_config[i];
		}
		return;
	}
	public static get_Weaponinfo(key){
		if(Weaponinfo.m_static_map.hasOwnProperty(key) == false){
			Weaponinfo.m_static_map[key] = Weaponinfo.create_Weaponinfo(key);
		}
		return Weaponinfo.m_static_map[key];	
	}
	public static create_Weaponinfo(key){
		if(weaponinfo_map.hasOwnProperty(key)){
			return new Weaponinfo(key);
		}
		return null;	
	}
	public static get_cfg_object():Object
	{
		return weaponinfo_map;
	}
}
}

File: tool/table/server/weaponinfo.py
# -*- coding: utf-8 -*-

'''
Author: Hannibal
Data: 
Desc: local data config
NOTE: Don't modify this file, it's build by xml-to-python!!!
'''


weaponinfo_map = {};
weaponinfo_map[1] = {"id":1,"aid":10001,};
weaponinfo_map[2] = {"id":2,"aid":10002,};
weaponinfo_map[3] = {"id":3,"aid":10003,};
weaponinfo_map[4] = {"id":4,"aid":10004,};
weaponinfo_map[5] = {"id":5,"aid":10005,};
weaponinfo_map[6] = {"id":6,"aid":10006,};
weaponinfo_map[7] = {"id":7,"aid":10007,};
weaponinfo_map[8] = {"id":8,"aid":10008,};
weaponinfo_map[9] = {"id":9,"aid":10009,};
weaponinfo_map[10] = {"id":10,"aid":10010,};
weaponinfo_map[11] = {"id":11,"aid":10011,};
weaponinfo_map[12] = {"id":12,"aid":10012,};
weaponinfo_map[13] = {"id":13,"aid":10013,};
weaponinfo_map[14] = {"id":14,"aid":10014,};
weaponinfo_map[15] = {"id":15,"aid":10015,};
weaponinfo_map[16] = {"id":16,"aid":10016,};
weaponinfo_map[17] = {"id":17,"aid":10001,};
weaponinfo_map[18] = {"id":18,"aid":10001,};
weaponinfo_map[19] = {"id":19,"aid":10001,};
weaponinfo_map[20] = {"id":20,"aid":10001,};
weaponinfo_map[21] = {"id":21,"aid":10001,};
weaponinfo_map[128] = {"id":128,"aid":10017,};
weaponinfo_map[129] = {"id":129,"aid":10018,};
weaponinfo_map[130] = {"id":130,"aid":10019,};
weaponinfo_map[131] = {"id":131,"aid":10020,};
weaponinfo_map[132] = {"id":132,"aid":10020,};
weaponinfo_map[133] = {"id":133,"aid":10020,};


class Weaponinfo:
	def __init__(self, key):
		config = weaponinfo_map.get(key);
		for k, v in config.items():
			setattr(self, k, v);
		return

def create_Weaponinfo(key):
		config = weaponinfo_map.get(key);
		if not config:
			return
		return Weaponinfo(key)

